### PR TITLE
feat(skeleton-loader): new component

### PR DESCRIPTION
**Pull request by saiponnada**
## Description

Added new skeleton loader component. 

Note:
Snapshots for the tests are not being generated.

## Context
#1976 

## Screenshots
image
image

### CODE REVIEW

File: src/components/ebay-skeleton-loader/index.marko
Comment (LuLaValva) [inline src/components/ebay-skeleton-loader/index.marko:16]: I think this is a sign that we either need to change `type` to `btn` or the class name to `skeleton--button`

File: src/components/ebay-skeleton-loader/index.marko
Comment (LuLaValva) [inline src/components/ebay-skeleton-loader/index.marko:15]: I'm not sure how I feel about defaulting to `"button"`, is that the most likely object to be making a skeleton for?

File: src/components/ebay-skeleton-loader/index.marko
Comment (LuLaValva) [inline src/components/ebay-skeleton-loader/index.marko:25]: Using `if` here means that we need to use JavaScript to create and destroy DOM nodes, which means that layout shift is likely and also a lot of work is being done each time a skeleton is shown or hidden. I think it would probably be preferable to toggle `style="visibility:hidden"` on the div instead.

File: src/components/ebay-skeleton-loader/index.marko
Comment (LuLaValva) [inline src/components/ebay-skeleton-loader/index.marko:36]: This is a lot of complex logic for the `tile` case, maybe it's a reason to move `ebay-skeleton-tile` into a separate component which has consistent logic (not sure about this comment, just a thought)

File: src/components/ebay-skeleton-loader/skeleton-loader.stories.js
Comment (saiponnada) [reply src/components/ebay-skeleton-loader/skeleton-loader.stories.js:38]: Was contemplating about that. I personally don't have any strong preference but felt unnecessary. Do we need to do that as we are listing all the stories?

File: src/components/ebay-skeleton-loader/index.marko
Comment (saiponnada) [reply src/components/ebay-skeleton-loader/index.marko:16]: This was tricky, the whole skin docs refers button as btn. I wish we named btn as button. Not sure why we did it. To keep it consistent, I can change type to `btn` but I really like type to be spelled out as button.

File: src/components/ebay-skeleton-loader/skeleton-loader.stories.js
Comment (LuLaValva) [reply src/components/ebay-skeleton-loader/skeleton-loader.stories.js:38]: Yes! You should include it every time, just so developers can see at a glance which attributes are being used.

File: src/components/ebay-skeleton-loader/index.marko
Comment (saiponnada) [reply src/components/ebay-skeleton-loader/index.marko:15]: The idea was to restrict to one of these types from ebay UI side. And picked button as random. We can also not load a skeleton if it doesn't match any type.

Comment (LuLaValva) [reply src/components/ebay-skeleton-loader/index.marko:15]: I would imagine that a skeleton without a type just adds `skeleton` to the class list, and nothing else

File: src/components/ebay-skeleton-loader/index.marko
Comment (saiponnada) [reply src/components/ebay-skeleton-loader/index.marko:25]: The idea for using `isVisible` is to set the flag to false to remove skeleton from dom and show content below. This is useful technique for hiding sibling skeleton. Preferably teams use other techniques like if-else or await placeholder than this approach. If we use `visibility:hidden`, won't the empty layout still visible beside the regular content if not used content swap approach?

Comment (agliga) [reply src/components/ebay-skeleton-loader/index.marko:25]: Yes we should keep it like this. Visibility hidden wont work
At the same time the tricky part of layout shift isnt hiding elements but ensuring that everything is the same size

File: src/components/ebay-skeleton-loader/index.marko
Comment (agliga) [reply src/components/ebay-skeleton-loader/index.marko:15]: Agreed. Unless its broken then we shouldn't default to that

File: src/components/ebay-skeleton-loader/index.marko
Comment (agliga) [reply src/components/ebay-skeleton-loader/index.marko:16]: I believe this was because of marko 3. (That or following bootstrap)
We can change it to button next major if needed. 
Lets keep it like this for now though.

File: src/components/ebay-skeleton-loader/skeleton-loader.stories.js
Comment (agliga) [reply src/components/ebay-skeleton-loader/skeleton-loader.stories.js:38]: We need type especially for booleans or enums.

Comment (saiponnada) [reply src/components/ebay-skeleton-loader/skeleton-loader.stories.js:38]: Done.

File: src/components/ebay-skeleton-loader/index.marko
Comment (agliga) [reply src/components/ebay-skeleton-loader/index.marko:36]: I think the right way is to make this component callable from others. 
Lets leave it as is for now. In future we should have button or tile having a "loading" option which would swap it to a skeleton component.

File: src/components/ebay-skeleton-loader/index.marko
Comment (LuLaValva) [reply src/components/ebay-skeleton-loader/index.marko:25]: Wait I still disagree with this at a fundamental level. Right now all `isVisible` does is control a top-level if statement, so really we should just remove this entirely and have teams use
```marko
<if(isVisible)>
  <ebay-skeleton-loader />
</if>
```

File: src/components/ebay-skeleton-loader/index.marko
Comment (saiponnada) [reply src/components/ebay-skeleton-loader/index.marko]: Done. renamed this to `skeleton-button`.

File: src/components/ebay-skeleton-loader/index.marko
Comment (saiponnada) [reply src/components/ebay-skeleton-loader/index.marko]: As discussed, now defaulted to form as we do not have standalone `skeleton` class.

File: src/components/ebay-skeleton-loader/index.marko
Comment (saiponnada) [reply src/components/ebay-skeleton-loader/index.marko]: yeah, for now I removed the wrapper condition and `isVisible`. Instead let's use documentation to explain the right techniques.

File: src/components/ebay-skeleton/index.marko
Comment (LuLaValva) [inline src/components/ebay-skeleton/index.marko:37]: I don't know if I like having this as an `input` option, because composite components won't look the same for every use case. It would be nice as an example, but I don't think it should be baked into the component like this.

Comment (saiponnada) [reply src/components/ebay-skeleton/index.marko:37]: No longer a component. Moved to composite example.

File: src/components/ebay-skeleton/index.marko
Comment (saiponnada) [reply src/components/ebay-skeleton/index.marko:35]: Regarding point 1, IMO majority of eBay users use English and, adding a consistent a11y text makes it consistent experience rather than not passing at all. All our components provide similar consistent experience for non-English app teams to pass a localized value, so hopefully it makes easier to adopt. Point 2 is taken care of with current code refactor as we are always providing the wrapper and adding a11yText on the wrapper.

File: src/components/ebay-skeleton/examples/withContent.marko
Comment (agliga) [inline src/components/ebay-skeleton/examples/withContent.marko:29]: Id rename this more specific. I believe this might be used by storybook or in other places. 
Maybe `skeleton-example-container`

File: src/components/ebay-skeleton/skeleton.stories.js
Comment (agliga) [inline src/components/ebay-skeleton/skeleton.stories.js:220]: I have a shortcut way of doing all this. 
https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-button/button.stories.js#L224-L227

The third argument is args if needed. 
https://github.com/eBay/ebayui-core/blob/336467022d6d755e714e48b90e48aefc4d9bca0d/.storybook/utils.js#L24

**Review by agliga — COMMENTED** [review]
Overall LGTM.

File: src/components/ebay-skeleton/examples/withContent.marko
Comment (saiponnada) [reply src/components/ebay-skeleton/examples/withContent.marko:29]: Done, renamed consistent to other story class names.

File: src/components/ebay-skeleton/skeleton.stories.js
Comment (saiponnada) [reply src/components/ebay-skeleton/skeleton.stories.js:220]: Done.

File: src/components/ebay-skeleton/components/ebay-skeleton-avatar/index.marko
Comment (agliga) [inline src/components/ebay-skeleton/components/ebay-skeleton-avatar/index.marko:3]: Lets use the new syntax for this
Its somethibg like
```
$ const { as, class: className, ...rest} = input
<div class=className processHtmlAttributes(...rest)>
```
Check with Luke for the right syntax.

**Review by agliga — COMMENTED** [review]
Lgtm overall
One comment. Otherwise its good to go imo

**Review by LuLaValva — APPROVED** [review]
Awesome work, thanks for making all of the modifications

File: src/components/ebay-skeleton/components/ebay-skeleton-avatar/index.marko
Comment (LuLaValva) [reply src/components/ebay-skeleton/components/ebay-skeleton-avatar/index.marko:3]: This will be done before my refactor, so I think it's ok to merge now and I'll do this change after a rebase.

**Review by agliga — APPROVED** [review]

**Pull request merged**
